pkg/virtual/workspaces/registry/org.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package registry
+
+import (
+	"time"
+
+	rbacinformers "k8s.io/client-go/informers/rbac/v1"
+	rbacv1client "k8s.io/client-go/kubernetes/typed/rbac/v1"
+	rbacv1listers "k8s.io/client-go/listers/rbac/v1"
+
+	tenancyclient "github.com/kcp-dev/kcp/pkg/client/clientset/versioned/typed/tenancy/v1alpha1"
+	workspaceinformer "github.com/kcp-dev/kcp/pkg/client/informers/externalversions/tenancy/v1alpha1"
+	frameworkrbac "github.com/kcp-dev/kcp/pkg/virtual/framework/rbac"
+	workspaceauth "github.com/kcp-dev/kcp/pkg/virtual/workspaces/auth"
+)
+
+func NewOrg(orgRBACClient rbacv1client.RbacV1Interface, orgClusteWorkspaceClient tenancyclient.ClusterWorkspaceInterface, orgRBACInformers rbacinformers.Interface, orgCRBInformer rbacinformers.ClusterRoleBindingInformer, orgClusterWorkspaceInformer workspaceinformer.ClusterWorkspaceInformer) Org {
+	orgSubjectLocator := frameworkrbac.NewSubjectLocator(orgRBACInformers)
+	orgReviewerProvider := workspaceauth.NewAuthorizerReviewerProvider(orgSubjectLocator)
+
+	orgWorkspaceAuthorizationCache := workspaceauth.NewAuthorizationCache(
+		orgClusterWorkspaceInformer.Lister(),
+		orgClusterWorkspaceInformer.Informer(),
+		orgReviewerProvider.ForVerb("get"),
+		orgRBACInformers,
+	)
+
+	var stopCh chan struct{}
+
+	orgWorkspaceAuthorizationCache.Run(1*time.Second, stopCh)
```

Comment:
this func is no constructor aka `New*` func. It is NewAndRun.

Reply:
would split both

Reply:
I agree with renaming it, to `CreateAndStartOrg` for example, but I'd rather keep a single method and thus ensure that when we get an Org instance, it *is* already running, until we stop it with the `Stop` method.

Reply:
renamed in commit c48becc80ef541475a0e032a26cfa6a9024c5ac8